### PR TITLE
WIP: Polar-aura: P1 Analytics Events

### PR DESCRIPTION
**Pull request by clottman**
## Links
* polar-aura.glitch.me
* https://app.clubhouse.io/glitch/story/4563/community-p1-events

## Changes:
* Connected to the WebDev source in Segment
* Includes events listed as P1 from the spreadsheet
* Removes all the old events
* Removes AnalyticsContext as the new analytics schema doesn't require this nor the origin property
* Renames TrackedButtonGroup to just ButtonGroup because we don't need to track the things the buttons were doing any more

## How To Test:
* Refer to the spreadsheet linked in the clubhouse ticket. Dedicated QA time will be held next week. 

## Feedback I'm looking for:
- Did the assumptions I made here match the intended functionality of the events? I'm worried some context got lost in the discussion around what various properties are supposed to represent. 

## Things left to do before deploying:
- [x] Change the source for events back to the prod source
- [x] Add in the CTA events (still waiting on a list of what the relevant calls to action are)

### CODE REVIEW

File: src/presenters/pages/router.js [hunk 1]
```diff
@@ -47,9 +47,18 @@ function ExternalPageReloader() {
...
 function track() {
   try {
-    const { analytics } = window;
+    const { analytics, location } = window;
     if (analytics) {
-      analytics.page({}, { groupId: '0' });
+      const searchParams = new URLSearchParams(location.search);
+      const properties = {
+        search: location.search.slice(1),
+        utmMedium: searchParams.get('utm_medium'),
```
Comment (sheridanvk) [inline src/presenters/pages/router.js]: I noticed that utm fields are automatically collected by Segment https://segment.com/docs/connections/spec/common/#context-fields-automatically-collected (you can see for yourself by passing in utm_campaign, etc, and look at the page event that gets sent by Segment; it includes all the info in the campaign section of the request body). It seems that these are also passed along to e.g. RedShift and such https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#context--traits. Do you think we could test this to see if we don't need to add this additional code and Segment just takes care of it for us?

Comment (clottman) [reply src/presenters/pages/router.js]: Good catch - removed the manual setters.

File: src/presenters/pages/create.js [hunk 2]
```diff
@@ -343,6 +354,15 @@ function ScreencapSection({ title, description, video, smallVideos, blob, image,
 }
 
 function Help() {
+  const { location } = useGlobals();
+  const helpCenterHref = 'https://glitch.com/help';
```
Comment (sarahzinger) [inline src/presenters/pages/create.js]: unrelated but maybe this should be a relative link

**Comment by clottman** [comment]
The shrinkwrap.yaml.swp file is junk, I think it's created when you try to open shrinkwrap.yaml in the glitch editor in a console-based text editor while the container is writing to that file? 

User ID is set because we called `window.analytics.identify` in current-user.js. 

The `analytics.page` stuff is kinda black-boxy to me; [the section on Default Properties (scroll down) in the docs](https://segment.com/docs/connections/sources/catalog/libraries/website/javascript/#page) talks about what properties are set by default.

**Comment by sarahzinger** [comment]
cool :) thanks for the explanation!

File: src/models/project.js [hunk 3]
```diff
@@ -74,3 +74,22 @@ export function userIsOnlyProjectAdmin({ project, user }) {
   if (adminCount > 1) return false;
   return userIsProjectAdmin({ project, user });
 }
+
+// To be kept in sync with source/data/base-project/domains.js in the editor repo
+const baseProjectIds = Object.freeze([
```
Comment (sarahzinger) [inline src/models/project.js]: TIL about .freeze thanks!

File: views/index.ejs [hunk 4]
```diff
@@ -138,8 +138,11 @@
         // js failed to parse, odds are the css would be broken too
         showFallback('noscript');
       }
-      // TODO change me before deploying 
-      analytics.load("Oa74eP053IozA1ouL7my4svHW36qGGmM");
+      if (window.location.hostname === "glitch.com") {
+        analytics.load("N8NetCAuTbHYGqtQtb44FFBSPUXwsAu2"); /* WebProd source in Segment */
+      } else {
+        analytics.load("Oa74eP053IozA1ouL7my4svHW36qGGmM"); /* WebDev source in Segment */
```
Comment (sarahzinger) [inline views/index.ejs]: this is cool! so from now on we should look at dev segment for remixes yeah?

Comment (sarahzinger) [reply views/index.ejs]: it looks like hostname is not supported on opera https://developer.mozilla.org/en-US/docs/Web/API/Location but maybe that's fine? or we could use .host? seems not that important probably

Comment (clottman) [reply views/index.ejs]: Yep dev segment will have the data for remixes and localhost. 

Great catch on hostname not being supported, I looked at that page but didn't see it :( Unfortunately it looks like host isn't supported either. How would you feel about defaulting to the prod source if `!window.location.hostname` in addition to the current condition? I don't think we'll ever be doing development on opera, and if we are I think analytics impact will be minimal.

File: src/models/project.js (hunk 3 above)
Comment (clottman) [reply src/models/project.js]: All thanks to Sheridan, I copied from the editor side. :)

File: views/index.ejs (hunk 4 above)
Comment (sarahzinger) [reply views/index.ejs]: oh dang that browser compatibility chart on mozilla is not consistent weird! on the location page it says host is supported, but on the host page it says it isn't :(

yeah your solution sounds perfect to me!

**Pull request merged**